controllers/devworkspace/state.go

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) 2019-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package devworkspace
+
+import (
+	"context"
+	"time"
+
+	orgv1 "github.com/eclipse-che/che-operator/api/v1"
+	chev2alpha1 "github.com/eclipse-che/che-operator/api/v2alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type DevWorkspaceState int
+
+const (
+	APINotPresentState DevWorkspaceState = 0
+	DisabledState      DevWorkspaceState = 1
+	EnabledState       DevWorkspaceState = 2
+)
+
+// ShouldDevWorkspacesBeEnabled evaluates if DevWorkspace mode should be enabled
+// which we only do, if there is the controller.devfile.io resource group in the cluster
+// and DevWorkspaces are enabled at least on one CheCluster
+func ShouldDevWorkspacesBeEnabled(mgr manager.Manager) (bool, error) {
```

Comment:
I would say it is more about `can` than `should`

Reply:
that's can and should )

Reply:
We should not care about multiple operators with devworkspace enabled. Should we?